File: VendingMachine/src/VendingMachine.js
class VendingMachine {
	constructor (newName, newLocation){
		this.name = newName
		this.location = newLocation
		this.allMyCoins = []
	}

	toString() {
		const result = `Vending Machine ${NEWLINE} ${TAB}${this.name} ${NEWLINE} ${TAB}${this.location}`
		return result
	}

    toTable(){
		let result = `<table>`
			result += `<tr> <th colspan="3">Vending Machine</th> </tr>`
			result += `<tr> <td>Name</td> <td>${this.name}</td> </tr>`
			result += `<tr> <td>Location</td> <td>${this.location}</td> </tr>`
			result += `</table>`

		return result
  }

	addCoin10cent(newAlloy, newDiameter, newWeight, newValue){
		var aCoin = new Coin(newAlloy, newDiameter, newWeight, newValue)
		this.allMyCoins.push(aCoin)
	}

	toStringCoin(){
		return `${this.allMyCoins[0].alloy}, ${this.allMyCoins[0].diameter}, ${this.allMyCoins[0].weight}, ${this.allMyCoins[0].value}`
	}

	toCoinTable(){
		let result = `<table>`
			result += `<tr> <th colspan="3">Coins</th> </tr>`
			result += `<tr> <td>Alloy</td> <td id='right'>${this.allMyCoins[0].alloy}</td> </tr>`
			result += `<tr> <td>Diameter</td> <td id='right'>${this.allMyCoins[0].diameter}</td> </tr>`
			result += `<tr> <td>Weight</td> <td id='right'>${this.allMyCoins[0].weight}</td> </tr>`
			result += `<tr> <td>Value</td> <td id='right'>${this.allMyCoins[0].value}</td> </tr>`
			result += `</table>`

		return result
	}
}

File: VendingMachine/pythonProject/vendingmachine.py
from Coin import Coin

class VendingMachine:
    def __init__(self, newName, newLocation):
        self.name = newName
        self.location = newLocation
        self.allMyCoins = []

    def __str__(self):
        result = f'{self.name} {self.location}'
        return result

    def to_table(self):
        result = f'<table><tr>'
        result += f'<td>{self.name}</td>'
        result += f'<td>{self.location}</td>'
        result += '</tr></table>'
        return result

    def add_coin(self, new_alloy, new_diameter, new_weight, new_value):
        new_coin = Coin(new_alloy, new_diameter, new_weight, new_value)
        self.allMyCoins.append(new_coin)

    def to_table_coin(self):
        result = f'<table>'
        result += f'<tr> <th colspan="2">Coins</th> </tr>'
        result += f'<tr> <td>Alloy</td> <td>{self.allMyCoins[0].alloy} </td> </tr>'
        result += f'<tr> <td>Diameter</td> <td>{self.allMyCoins[0].diameter}</td> </tr>'
        result += f'<tr> <td>Weight</td> <td>{self.allMyCoins[0].weight}</td> </tr>'
        result += f'<tr> <td>Value</td> <td>{self.allMyCoins[0].value}</td> </tr>'
        result += '</table>'
        return result
